File: scratch/search_white_bg.py
import os
import re

css_dir = "C:/laragon/www/hms-master/public/css"
assets_css_dir = "C:/laragon/www/hms-master/assets/css"

search_dirs = [css_dir, assets_css_dir]

bg_pattern = re.compile(r'(background|background-color)\s*:\s*(#[fF]{3,6}|white|rgba\(\s*255\s*,\s*255\s*,\s*255)', re.IGNORECASE)

matches = []

for s_dir in search_dirs:
    if not os.path.exists(s_dir):
        continue
    for root, dirs, files in os.walk(s_dir):
        for file in files:
            if file.endswith('.css'):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8', errors='ignore') as f:
                    content = f.read()
                    # Find lines with background-color or background setting white
                    lines = content.split('\n')
                    for idx, line in enumerate(lines):
                        if bg_pattern.search(line):
                            # Also check if it's related to tables
                            if any(k in line.lower() for k in ['table', 'tbody', 'tr', 'td', 'responsive', 'box', 'tab']):
                                matches.append((file, idx + 1, line.strip()))

print(f"Found {len(matches)} potential white background leaks:")
for m in matches[:100]:
    print(f"{m[0]}:{m[1]} -> {m[2]}")

File: scratch/search_white_bg.php
<?php
$search_dirs = [
    "C:/laragon/www/hms-master/public/css",
    "C:/laragon/www/hms-master/assets/css"
];

$matches = [];

foreach ($search_dirs as $dir) {
    if (!is_dir($dir)) continue;
    $iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($dir));
    foreach ($iterator as $file) {
        if ($file->isFile() && $file->getExtension() === 'css') {
            $path = $file->getPathname();
            $content = file_get_contents($path);
            if ($content === false) continue;
            
            $lines = explode("\n", $content);
            foreach ($lines as $idx => $line) {
                // Search for background or background-color setting to white/light color
                if (preg_match('/(background|background-color)\s*:\s*(#[fF]{3,6}|white|rgba\(\s*255\s*,\s*255\s*,\s*255)/i', $line)) {
                    $lower_line = strtolower($line);
                    if (strpos($lower_line, 'table') !== false || 
                        strpos($lower_line, 'tbody') !== false || 
                        strpos($lower_line, 'tr') !== false || 
                        strpos($lower_line, 'td') !== false || 
                        strpos($lower_line, 'responsive') !== false || 
                        strpos($lower_line, 'box') !== false || 
                        strpos($lower_line, 'tab') !== false) {
                        $matches[] = [
                            'file' => basename($path),
                            'line' => $idx + 1,
                            'content' => trim($line)
                        ];
                    }
                }
            }
        }
    }
}

$output = "Found " . count($matches) . " potential white background leaks:\n";
foreach ($matches as $m) {
    $output .= sprintf("%s:%d -> %s\n", $m['file'], $m['line'], $m['content']);
}

file_put_contents("C:/Users/USER/.gemini/antigravity/brain/76a581f7-3102-47fa-bd37-aae5eade9323/scratch/search_results.txt", $output);
echo "Written results to C:/Users/USER/.gemini/antigravity/brain/76a581f7-3102-47fa-bd37-aae5eade9323/scratch/search_results.txt\n";
?>
